tests/Tests.ServiceModel/ServiceModel/Discovery/AnnouncementServiceTests.cs

```diff
@@ -44,5 +51,58 @@ public void AnnounceEndpoints_does_stuff([Frozen] AnnouncementServiceOptions opt
 
             disposable.Dispose();
         }
+
+        [Test, CustomAutoData]
+        public void AnnounceEndpoints_suppress_exceptions_for_AnnounceOnline(
+            [Frozen] AnnouncementServiceOptions options,
+            IAnnouncementClientWrapper clientWrapper,
+            [Frozen] IAnnouncementClientWrapperFactory clientFactory,
+            ILogger<AnnouncementService> logger, ServiceEndpoint testEndpoint, Uri endpointUri)
+        {
+            options.EndpointDiscoveryMetadata = (_) => new EndpointDiscoveryMetadata();
+            options.Interval = TimeSpan.FromSeconds(1);
+            options.IsAnnouncementEnabled = true;
+
+            testEndpoint.Address = new EndpointAddress(endpointUri);
+            testEndpoint.Behaviors.Add(new AnnounceableBehavior());
+
+            var scheduler = new TestScheduler();
+
+            var sut = new AnnouncementService(new OptionsWrapper<AnnouncementServiceOptions>(options), logger, clientFactory, scheduler);
+
+            Mock.Get(clientWrapper).Setup(i => i.AnnounceOnline(It.IsAny<EndpointDiscoveryMetadata>())).Throws<Exception>();
+
+            Mock.Get(clientFactory).Setup(i => i.Create(It.IsAny<Uri>(), It.IsAny<Binding>())).Returns(clientWrapper);
+
+            Assert.DoesNotThrow(() => sut.AnnounceEndpoints(new[] {testEndpoint}));
+
+            scheduler.AdvanceBy(options.Interval.Ticks);
```

Comment:
You might get false positives here because the scheduler is advanced and the test terminates.

Reply:
True! Moved in scheduler call within the Assert. Not sure if it's safe enough though? Just been testing with removing/readding try catch in the announce and the test is red/green as it should be but not 100% sure with this. Did not find any perfect example of this so just tested with the scheduler.AdvanceBy solution